pkg/manifests/manifests_test.go

```diff
@@ -276,6 +278,131 @@ func TestUnconfiguredManifests(t *testing.T) {
 	}
 }
 
+func TestHTTPConfig(t *testing.T) {
+	type checkFunc func(*monv1.Alertmanager) error
+
+	checks := func(cs ...checkFunc) checkFunc {
+		return func(a *monv1.Alertmanager) error {
+			for _, f := range cs {
+				if err := f(a); err != nil {
+					return err
+				}
+			}
+			return nil
+		}
+	}
+
+	hasContainers := func(expected bool) checkFunc {
+		return func(a *monv1.Alertmanager) error {
+			if got := len(a.Spec.Containers) > 0; got != expected {
+				return fmt.Errorf("expected spec to have containers %t, got %t", expected, got)
+			}
+			return nil
+		}
+	}
+
+	containersHaveEnv := func(name, value string) checkFunc {
+		return func(a *monv1.Alertmanager) error {
+			for c := range a.Spec.Containers {
+				for e := range a.Spec.Containers[c].Env {
+					if a.Spec.Containers[c].Env[e].Name != name {
+						continue
+					}
+					if a.Spec.Containers[c].Env[e].Value == value {
+						return nil
+					}
+				}
+			}
+
+			return fmt.Errorf("containers expected to have env name %v=%v, but they don't", name, value)
```

Comment:
should `env name` here actually be `env var`?

Reply:
right :+1: it should actually be `name env=var`, I suggest simply to reduce to `containers expected to have env %v=%v, but they don't`

Reply:
ah, sorry, your suggestion actually makes perfect sense.